api/app/signals/apps/email_integrations/custom_backends/rest_email_endpoint.py

```diff
@@ -0,0 +1,41 @@
+import json
+
+from django.conf import settings
+from django.core.mail.backends.base import BaseEmailBackend
+from djcelery_email.utils import chunked, email_to_dict
+from requests import RequestException, Session
+
+
+class RestEmailBackend(BaseEmailBackend):
+    def __init__(self, fail_silently=False, **kwargs):
+        super(RestEmailBackend, self).__init__(fail_silently)
+        self.init_kwargs = kwargs
+
+    def _send_email_rest_api(self, session, message):
+        try:
+            response = session.post(
+                url=settings.EMAIL_REST_ENDPOINT,
+                headers={'Content-type': 'application/json', 'Accept': 'text/plain'},
+                data=json.dumps(message),
+                timeout=settings.EMAIL_REST_ENDPOINT_TIMEOUT,
+            )
+            response.raise_for_status()
+        except RequestException as e:
+            if not self.fail_silently:
+                raise e
+            else:
+                return False
+        return True
+
+    def send_messages(self, email_messages):
+        messages_sent = 0
+        s = Session()
+        # init session, connection using client and server side certificate is expensive
+        if hasattr(settings, 'EMAIL_REST_ENDPOINT_CLIENT_CERT') and hasattr(settings, 'EMAIL_REST_ENDPOINT_CLIENT_KEY'): # noqa
+            s.cert = (settings.EMAIL_REST_ENDPOINT_CLIENT_CERT, settings.EMAIL_REST_ENDPOINT_CLIENT_KEY)
+
+        for chunk in chunked(email_messages, settings.CELERY_EMAIL_CHUNK_SIZE):
+            chunk_messages = [email_to_dict(msg) for msg in chunk]
+            if self._send_email_rest_api(s, chunk_messages):
+                messages_sent += 1
```

Comment:
This is not actually counting the number of emails sent (in stead it counts the number of batches that were sent) `BaseEmailBackend.send_messages docstring`:
```
        """
        Send one or more EmailMessage objects and return the number of email
        messages sent.
        """
```

Reply:
this is actually a typo. `chunk_messages` should have been `chunk_message`. chunk message is the dict of all attribute of a msg (within a chunk). we rename to make more clear

Reply:
Point was:

Reply:
There was a bug, the _send() only sends a single message, so we should process the message chunks dicts. We cannot do += len() since a message can fail. I've corrected this